src/lib/web-component/filter-timestamp/filter-timestamp-element.js

```diff
@@ -75,6 +83,14 @@ export class FilterTimestamp extends HTMLElement {
     }
   }
 
+  setTimestampStatus() {
+    if (this.timeStamp != null || Object.keys(this.timeStamp).length !== 0) {
+      if (!this.classList.contains('expanded')) {
```

Comment:
This "expanded" class name is an implementation detail of DropDownController. To avoid having to check the `expanded` class here, let `DropDownController` have three methods:

- showDropDown
- hideDropDown
- toggleDropDown.

... and call the right one from the right location.

Triggering a fake `click` event from here looks strange too. You should directly invoke the `dropDownController` instead.

Reply:
Made these 3 methods in `DropDownController`. Used the `showDropDown` method in this case.